File: backend/events/listEvents.mjs
import { ScanCommand } from "@aws-sdk/lib-dynamodb";
import { db, EVENTS_TABLE } from "./shared/db.mjs";
import { ok, serverError, cors } from "./shared/response.mjs";

export async function handler(event) {
  if (event.httpMethod === "OPTIONS") return cors();

  try {
    const result = await db.send(new ScanCommand({ TableName: EVENTS_TABLE }));
    const events = (result.Items ?? []).sort(
      (a, b) => new Date(b.createdAt) - new Date(a.createdAt)
    );
    return ok(events);
  } catch (e) {
    console.error(e);
    return serverError();
  }
}

File: backend/events/listEvents.py
import logging

from shared.db import events_table
from shared.response import cors, ok, server_error, with_cors

log = logging.getLogger()


@with_cors
def handler(event, context):
    if event.get("httpMethod") == "OPTIONS":
        return cors()

    try:
        result = events_table().scan()
        events = sorted(result.get("Items", []),
                        key=lambda e: e.get("createdAt") or "", reverse=True)
        return ok(events)
    except Exception:
        log.exception("listEvents failed")
        return server_error()
